### PR TITLE
fix: throws DatabaseException in DB connections

### LINKED ISSUE #6007: Bug: Different DB drivers have different types of Exceptions
Ref #4331

When `DBDebug` is true. 
If you set it false, the Exception will be gone and the query returns `false`.

MySQLi:
```
1) CodeIgniter\Database\Live\DbErrorTest::testForeignKeyConstraintFailsDBDebugTrue
mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`test`.`db_auth_identities`, CONSTRAINT `db_auth_identities_user_id_foreign` FOREIGN KEY (`user_id`) REFERENCES `db_auth_users` (`id`) ON DELETE CASCADE)
```
SQLite3:
```
1) CodeIgniter\Database\Live\DbErrorTest::testForeignKeyConstraintFailsDBDebugTrue
ErrorException: SQLite3::exec(): FOREIGN KEY constraint failed
```
Postgre:
```
1) CodeIgniter\Database\Live\DbErrorTest::testForeignKeyConstraintFailsDBDebugTrue
ErrorException: pg_query(): Query failed: ERROR:  insert or update on table "db_auth_identities" violates foreign key constraint "db_auth_identities_user_id_foreign"
DETAIL:  Key (user_id)=(2) is not present in table "db_auth_users".
```
OCI8:
```
 1) CodeIgniter\Database\Live\DbErrorTest::testForeignKeyConstraintFailsDBDebugTrue
ErrorException: oci_execute(): ORA-00942: table or view does not exist
```
SQLSRV:
```
 1) CodeIgniter\Database\Live\DbErrorTest::testForeignKeyConstraintFailsDBDebugTrue
Exception: [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]The INSERT statement conflicted with the FOREIGN KEY constraint "db_auth_identities_user_id_foreign". The conflict occurred in database "test", table "dbo.db_auth_users", column 'id'.
```

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes #6007

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by MGatner — COMMENTED** [review]
Looking good so far. Is the "breaking change" that people will have to change the type of exception they are catching (assuming they specifying)?

**Comment by kenjis** [comment]
> Is the "breaking change" that people will have to change the type of exception they are catching (assuming they specifying)?

Yes.

Before:
`mysqli_sql_exception` (extends `RuntimeException`)
`ErrorException` (extends `Exception`)
`Exception`
`ArgumentCountError`  (extends `TypeError`)

After:
`DatabaseExcetpion` (extends `Error`)

**Comment by kenjis** [comment]
I don't know why `DatabaseExcetpion` extends `Error`.

**Comment by MGatner** [comment]
> I don't know why `DatabaseExcetpion` extends `Error`.

I'm afraid I don't know. I have never been very active in the database layer, and it was almost entirely written before I joined the project. Maybe @lonnieezell remembers?

**Comment by kenjis** [comment]
Added the documentation.

File: system/Database/SQLSRV/PreparedQuery.php [hunk 1]
```diff
@@ -34,6 +35,18 @@ class PreparedQuery extends BasePreparedQuery
...
+     * A reference to the db connection to use.
+     *
+     * @var Connection
+     */
+    protected $db;
+
+    public function __construct(Connection $db)
+    {
+        parent::__construct($db);
+    }
```
Comment (MGatner) [inline system/Database/SQLSRV/PreparedQuery.php:48]: Is this to ensure a certain parameter type?

**Review by MGatner — APPROVED** [review]

File: system/Database/SQLSRV/PreparedQuery.php (hunk 1 above)
Comment (kenjis) [reply system/Database/SQLSRV/PreparedQuery.php:48]: Yes.
I added `getAllErrorMessages()` as `@internal` in `system/Database/SQLSRV/Connection.php`,
and use it.

**Pull request merged**